app/forms/questionnaire_form.py

```diff
@@ -109,25 +126,25 @@ def validate_date_range_question(self, question):
 
         return True
 
-    def validate_calculated_question(self, question):
+    def validate_calculated_question(self, question: QuestionSchema) -> bool:
         for calculation in question["calculations"]:
-            target_total, currency = self._get_target_total_and_currency(
-                calculation, question
-            )
-            if self.answers_all_valid(
-                calculation["answers_to_calculate"]
-            ) and self._validate_calculated_question(
-                calculation, question, target_total, currency
-            ):
-                # Remove any previous question errors if it passes this OR before returning True
-                if question["id"] in self.question_errors:
-                    self.question_errors.pop(question["id"])
-                return True
+            result = self._get_target_total_and_currency(calculation, question)
+            if result:
+                target_total, currency = result
+                if self.answers_all_valid(
+                    calculation["answers_to_calculate"]
+                ) and self._validate_calculated_question(
+                    calculation, question, target_total, currency
+                ):
+                    # Remove any previous question errors if it passes this OR before returning True
+                    if question["id"] in self.question_errors:
+                        self.question_errors.pop(question["id"])
+                    return True
```

Comment:
Not a blocker, but usually it is nice to exit early to avoid such nesting to improve readability.

Reply:
I'm not sure we'd be exiting early from the loop because every calculation in the loop needs to be checked. I could do this to remove the larger nested block?
```
            if not result:
                continue
```

Reply:
Sorry by exit, that was what I was referring to in the instance.